crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4084,16 +4084,19 @@ impl<'db> TypeInferenceBuilder<'db> {
         // https://typing.readthedocs.io/en/latest/spec/annotations.html#grammar-token-expression-grammar-type_expression
 
         match expression {
-            ast::Expr::Name(name) => {
-                debug_assert_eq!(name.ctx, ast::ExprContext::Load);
-                self.infer_name_expression(name).to_instance(self.db)
-            }
+            ast::Expr::Name(name) => match name.ctx {
+                ast::ExprContext::Load => self.infer_name_expression(name).to_instance(self.db),
+                ast::ExprContext::Invalid => Type::Unknown,
+                ast::ExprContext::Store | ast::ExprContext::Del => Type::Todo,
```

Comment:
We could also go with
```suggestion
                ast::ExprContext::Store | ast::ExprContext::Del => todo!(),
```
here (and below) for now, if we want to be more explicit about the fact that we can't handle this yet.

Reply:
I prefer your current version — I think we want to be able to check as much code as possible without false positives right now... And you can think of a panic as just an exceptionally dramatic false positive 😆